contracts/Colony.sol

```diff
@@ -74,12 +74,54 @@ contract Colony is ColonyStorage, PatriciaTreeProofs {
     setRoleAssignmentFunction(bytes4(keccak256("setTaskEvaluatorRole(uint256,address)")));
     setRoleAssignmentFunction(bytes4(keccak256("setTaskWorkerRole(uint256,address)")));
 
+    // Setting this so there will be no tokens available for issuing when the colony is created
+    totalIssuedInSeconds = now;
+
     // Initialise the root domain
     IColonyNetwork colonyNetwork = IColonyNetwork(colonyNetworkAddress);
     uint256 rootLocalSkill = colonyNetwork.getSkillCount();
     initialiseDomain(rootLocalSkill);
   }
 
+  function setTokenIssuanceRate(uint256 _amount, uint256 _period, uint256 _precision) public auth {
+    require(_amount > 0 && _period > 0, "token-issuance-rate-bad-inputs");
+    require(sub(now, lastIssuenceRateChangeTimestamp) >= 4 weeks, "token-issuance-rate-change-not-allowed");
+
+    // If we are setting the rate for the first time we will not go through these checks
+    if (tokenIssuanceRate.amount > 0 && tokenIssuanceRate.period > 0) {
+      // Calculate the current rate in form of token/second
+      uint256 oldRate = mul(tokenIssuanceRate.amount, _precision) / tokenIssuanceRate.period;
+      // Calculate new rate
+      uint256 newRate = mul(_amount, _precision) / _period;
+      // this will throw if `_precision` is not high enough
+      require(oldRate > 0 && newRate > 0, "token-issuance-rate-bad-precision");
+      // Get the difference between rates
+      uint256 diff = (oldRate > newRate) ? sub(oldRate, newRate) : sub(newRate, oldRate);
+      require(diff > 0, "token-issuance-rate-invalid-rate");
+      // How much is new rate different (in percent) relative to the current one
+      uint256 percent = mul(diff, 100) / oldRate;
+      // Must be less than 10 percent
+      require(percent <= 10, "token-issuance-rate-change-too-big");
```

Comment:
For `a` and `p` are as defined in PR description this requirement can be simplified to the following
`11*a1*p2 >=10*a2*p1` where `a2/p2 > a1/p1` i.e. increasing issuance rate
`9*a1*p2 <= 10*a2*p1` where `a1/p1 > a2/p2` i.e. decreasing issuance rate
thus removing the need for `precision` and other calculations. 

To protect against overflows I'd suggest changing the `uint256` types of `tokenIssuanceRate.amount` to `uint128` and `tokenIssuanceRate.period` to `uint32`. 
`uint128` => upper limit = `340282366920938463463374607431768211455` token wei
`uint32` => upper limit = `((2^32-1)/2628000)/12 = 136.19 years` which is more than sufficient. 

It should provide sufficient range of values yet prevent overflows and also save 20,000 gas and an extra slot per `TokenIssuanceRate` entry.

Reply:
Following a conversation with @area he suggested to fix the issuance rate period to say a month and only allow the `amount` to vary. This would limit to some negligible extent the issuance rate variations to min `30*24*3600=2592000` tokens per month which considering it is significantly less than `10^18` is deemed acceptable.
Can you think of any issues with that model? Perhaps plug this (even simpler) version in the suggestion above.